### DISCUSSION TITLE
[Blog] Diagram Driven Cloud Infrastructure?

### DISCUSSION THREAD

**Discussion by lukaszgasior** [AzureCommunityPL/group#80] [Community Content]
Hey,

znalazłem ciekawe narzędzie do tworzenia infrastruktury w chmurze - https://gasior.net.pl/blog/diagram-driven-cloud-infrastructure-with-cloud-maker/

Rysujecie diagram ... i robicie deployment. Bez ARM templatów, terraforma, powershella czy innego CLI.
Co o myślicie o takim podejściu?

**Comment by kamil-mrzyglod** [comment]
Wygląda fajnie, ale ja zawsze mam problem z takimi narzędziami :) Trochę jak z low-code - świetne rozwiązanie dla osób, które chcą stworzyć coś do biznesu bez angażowania całego działu IT, pytania jednak pojawiają się później:

- jak kwestie licencji kiedy robisz handover do operations
- jaki problem tak naprawdę WYSIWYGiem rozwiązujesz
- co z kwestiami walidacji / polityk, które aplikujesz na poziomie samego kodu
- co ze wsparciem funkcjonalności w preview
- jak ze wsparciem różnych wersji API
- jak z debuggowaniem wygenerowanego kodu deploymentu(i poziom wsparcia ze strony dostawcy)
- jak ze wsparciem deploymentu na róznych scope'ach
- ...

I najważniejsze pytanie - jak z rysowaniem diagramów kiedy masz więcej niż 5 zasobów w ramach szablonu ;)

**Comment by wi5nia** [comment]
Podejscie oczywiscie mega spoko, natomiast bylo juz sporo prob zrobienia czegos takiego i sie rozbijalo zazwyczaj na utrzymaniu zgodnosci z nowymi wersjami API czy komponenty jak mowil @kamil-mrzyglod 

Mysle jednak, ze [Bicep](https://github.com/azure/bicep) moze duzo poprawic i juz pisanie tego z palca nie bedzie takie straszne bo demo tydzien temu pokazalo fajny kierunek, w ktorym to idzie.

**Comment by lukaszgasior** [comment]
oni z diagramu generują ARMa, z którego potem robią deployment - więc na poziomie danej wersji ARMa to powinno działać, choć pewnie będzie problem z nowościami

sam jestem ciekaw jak takie narzędzie sprawdziłoby się w realnym projekcie, do zabawy to nawet spoko jest

a bicep wygląda przyjemnie 👍

**Comment by kamil-mrzyglod** [comment]
Jest jeszcze Diagram-as-Code ;) https://github.com/mingrammer/diagrams
